Content.Client/Fax/UI/FaxBoundUi.cs

```diff
@@ -22,12 +29,35 @@ protected override void Open()
         _window.OpenCentered();
 
         _window.OnClose += Close;
+        _window.FileButtonPressed += OnFileButtonPressed;
         _window.CopyButtonPressed += OnCopyButtonPressed;
         _window.SendButtonPressed += OnSendButtonPressed;
         _window.RefreshButtonPressed += OnRefreshButtonPressed;
         _window.PeerSelected += OnPeerSelected;
     }
 
+    private async void OnFileButtonPressed()
+    {
+        if (_dialogIsOpen)
+            return;
+        var filters = new FileDialogFilters(new FileDialogFilters.Group("txt"));
+        await using var file = await _fileDialogManager.OpenFile(filters);
+        _dialogIsOpen = true;
+
+        if (_window == null || _window.Disposed || file == null)
+        {
+            _dialogIsOpen = false;
+            return;
+        }
+
+        var reader = new StreamReader(file);
+        var content = await reader.ReadToEndAsync();
+        SendMessage(new FaxFileMessage(content.Substring(0, Math.Min(content.Length, FaxFileMessageValidation.MaxContentSize)), _window.OfficePaper));
+        
+        reader.Close();
+        _dialogIsOpen = false;
+    }
+
```

Comment:
You have pushes from maintainers off so I'm just going to do it here.

```suggestion
    private async void OnFileButtonPressed()
    {
        if (_dialogIsOpen)
            return;
            
        _dialogIsOpen = true;
        var filters = new FileDialogFilters(new FileDialogFilters.Group("txt"));
        await using var file = await _fileDialogManager.OpenFile(filters);
        _dialogIsOpen = false;

        if (_window == null || _window.Disposed || file == null)
        {
            return;
        }

        using var reader = new StreamReader(file);
        var content = await reader.ReadToEndAsync();
        SendMessage(new FaxFileMessage(content[..Math.Min(content.Length, FaxFileMessageValidation.MaxContentSize)], _window.OfficePaper));
    }

```